### PR TITLE
Restore sll error behaviour

### PR DESCRIPTION
**Pull request by TheOneRing**

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests</strong> https://drone.owncloud.com/owncloud/client/11760/6/1

:boom: The GUI tests failed.

GUI Logs: https://cache.owncloud.com/public/owncloud/client/11760/guiReportUpload/index.html

<details>

Server Logs: https://cache.owncloud.com/public/owncloud/client/11760/guiReportUpload/serverlog.log

Screenshots:
- [Adding_account_with_self_signed_certificate_for_the_first_time.png](https://cache.owncloud.com/public/owncloud/client/11760/guiReportUpload/screenshots/Adding_account_with_self_signed_certificate_for_the_first_time.png)
- [Adding_account_with_wrong_credentials.png](https://cache.owncloud.com/public/owncloud/client/11760/guiReportUpload/screenshots/Adding_account_with_wrong_credentials.png)
- [Adding_multiple_accounts.png](https://cache.owncloud.com/public/owncloud/client/11760/guiReportUpload/screenshots/Adding_multiple_accounts.png)
- [Adding_normal_Account.png](https://cache.owncloud.com/public/owncloud/client/11760/guiReportUpload/screenshots/Adding_normal_Account.png)
- [Delete_a_file.png](https://cache.owncloud.com/public/owncloud/client/11760/guiReportUpload/screenshots/Delete_a_file.png)
- [Syncing_a_file_to_the_server.png](https://cache.owncloud.com/public/owncloud/client/11760/guiReportUpload/screenshots/Syncing_a_file_to_the_server.png)

</details>

File: src/libsync/accessmanager.cpp [hunk 1]
```diff
@@ -40,6 +42,16 @@ AccessManager::AccessManager(QObject *parent)
     setConfiguration(QNetworkConfiguration());
 #endif
     setCookieJar(new CookieJar);
+
+    connect(this, &AccessManager::sslErrors, this, [this](QNetworkReply *reply, const QList<QSslError> &errors) {
```
Comment (fmoc) [inline src/libsync/accessmanager.cpp:46]: Can we make sure this slot is run first using some flag on `connect`?

**Review by fmoc — APPROVED** [review]
Looks good. I'm not a fan of the old behavior, though. Is the idea of storing the error along with the certificate still something we want to realize (even if just long-term)?

**Comment by TheOneRing** [comment]
> Looks good. I'm not a fan of the old behavior, though. Is the idea of storing the error along with the certificate still something we want to realize (even if just long-term)?

Please open an issue for 4.0

File: src/libsync/accessmanager.cpp (hunk 1 above)
Comment (fmoc) [reply src/libsync/accessmanager.cpp:46]: Thinking about it, maybe we should just debug log all erased errors. That way, whoever will work with this code in the future, will get notified about ignored errors.

**Pull request merged**